src/main/java/com/github/kongchen/swagger/docgen/AbstractDocumentSource.java

```diff
@@ -159,7 +160,17 @@ public void toSwaggerDocuments(String uiDocBasePath, String outputFormats, Strin
     public void loadModelModifier() throws GenerateException, IOException {
         ObjectMapper objectMapper = Json.mapper();
         if (apiSource.isUseJAXBAnnotationProcessor()) {
-            objectMapper.registerModule(new JaxbAnnotationModule());
+            JaxbAnnotationModule jaxbAnnotationModule = new JaxbAnnotationModule();
+            if (apiSource.isUseJAXBAnnotationProcessorAsPrimary()) {
+                jaxbAnnotationModule.setPriority(Priority.PRIMARY);    
+            } else {
+                jaxbAnnotationModule.setPriority(Priority.SECONDARY);
+            }
+            objectMapper.registerModule(jaxbAnnotationModule);
+            
+            // to support @ApiModel on class level.
+            // must be registered only if we use JaxbAnnotationModule before. Why?
```

Comment:
I do not understand plugin or swagger-core internals well. 

1. I don't understand why this happens but when JAXB is disabled Swagger annotations are used. This is fine. But when I register JAXB module (as PRIMARY or SECONDARY) Swagger annotations are not used anymore. I had to register Swagger explicitly.
2. When I manually register default SwaggerModule `@ApiModel` is ignored. That's why I use enhanced SwaggerModule. But it works fine if JAXB module is not loaded. 

Confusing...